specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/compute.json

```diff
@@ -3517,6 +3517,464 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{restorePointCollectionName}": {
+      "put": {
+        "tags": [
+          "RestorePointCollections"
+        ],
+        "operationId": "RestorePointCollections_CreateOrUpdate",
+        "description": "The operation to create or update the restore point collection.",
+        "x-ms-examples": {
+          "Create an availability set.": {
+            "$ref": "./examples/CreateARestorePointCollection.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "restorePointCollectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the restore point collection."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RestorePointCollection"
+            },
+            "description": "Parameters supplied to the Create or Update restore point collection operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/RestorePointCollection"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/RestorePointCollection"
+            }
+          },
+          "default": {
```

Comment:
why default is the error response, Dont we need to mention only allowed http status codes for the successfull scenarios. Check other resources in the swagger for references.  Also  in which scenarios we return 200, I believe we always return 201 since this api is used for only create only but not update.

Reply:
1. Default is supposed to be error response. I followed the pattern in the rest of compute.json and they all have error as default. Also, if i remove default error, then the automatic validation fails saying I have  to include error.
2. we allow update of restore point collection (for RPC, update of tags is allowed) for which we return 200.
PUT RPC when creating RPC = 201
PUT RPC when updating RPC = 200

Reply:
I am not finding this default response in resources like dedicated host put and patch API specs. Can you check once. did find it in few resources like image, extensions . Are these specific to proxy only resources?

Reply:
Are these specific to proxy only resources? no. The only proxy only resource in compute.json is restore point.
Restore Point collection is top-level arm resource.
will investigate more tomorrow

Reply:
I removed the default error response just now, and the Lint validation immediately failed saying that default error response is needed. 

It looks like this requirement was added in May 2020.
https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#R4010
I am guessing that maybe all the operations added after May2020 have the default error added by dev owner?

Applies to : ARM OpenAPI(swagger) specs

Output Message : The response is defined without a default error response implementation,please add it.

Description : Per ARM Specs, every operation must have a default error response implementation.

CreatedAt: May 21, 2020

LastModifiedAt: May 21, 2020

How to fix the violation: For each operation response, please add a default error response implementation.

I am going to follow the ARM documentation and add the default error

Reply:
also, I looked at other ppls PRs and they are also adding CloudError as default
for ex: https://github.com/Azure/azure-rest-api-specs/pull/12442/files